specifyweb/frontend/js_src/lib/components/QueryBuilder/fieldSpec.ts

```diff
@@ -90,11 +107,15 @@ export class QueryFieldSpec {
     SpQueryField['fields'],
     'fieldName' | 'isRelFld' | 'stringId' | 'tableList'
   > {
+    this.overrideIsRelationship =
+      this.treeRank !== '-any' &&
```

Comment:
Don't hardcode `-` or `any` - we might change these in the future (and we did in the past) and so it would be easy forget this usage and create a bug.

Replace `-any` with `anyTreeRank`

Reply:
Fixed in 129d825